paasta_tools/instance/kubernetes.py

```diff
@@ -407,6 +407,96 @@ def filter_actually_running_replicasets(
     ]
 
 
+def kubernetes_status_v2(
+    service: str,
+    instance: str,
+    verbose: int,
+    include_smartstack: bool,
+    include_envoy: bool,
+    instance_type: str,
+    settings: Any,
+):
+    status: Dict[str, Any] = {}
+    config_loader = LONG_RUNNING_INSTANCE_TYPE_HANDLERS[instance_type].loader
+    job_config = config_loader(
+        service=service,
+        instance=instance,
+        cluster=settings.cluster,
+        soa_dir=settings.soa_dir,
+        load_deployments=True,
+    )
+    kube_client = settings.kubernetes_client
+    if kube_client is None:
+        return status
+
+    app_name = job_config.get_sanitised_deployment_name()
+    pod_list = kubernetes_tools.pods_for_service_instance(
+        service=job_config.service,
+        instance=job_config.instance,
+        kube_client=kube_client,
+        namespace=job_config.get_kubernetes_namespace(),
+    )
+    pods_by_replicaset = get_pods_by_replicaset(pod_list)
+    replicaset_list = kubernetes_tools.replicasets_for_service_instance(
+        service=job_config.service,
+        instance=job_config.instance,
+        kube_client=kube_client,
+        namespace=job_config.get_kubernetes_namespace(),
+    )
```

Comment:
Even though it's not widely used, we can have statefulsets instead of replicasets here, so we should really be sorting pods by either statefulset or replicaset. 

You can maybe punt on this. However, keep in mind that since the current paasta status doesn't connect replicasets/statefulsets to pods, it's okay to leave out statefulsets. But, since we want to connect it all in a table, punting would probably mean not showing any table at all.

Reply:
hmm yeah good catch -- I will either fix this or file a follow-up

Reply:
Added a TODO